File: parse_faq.js
const fs = require('fs');
const content = fs.readFileSync("/Users/ishtarpissano/.gemini/antigravity/brain/623961af-513c-49e6-b31d-d37c2d1ecdc4/.system_generated/steps/408/content.md", "utf8");
const match = content.match(/<script id="__NEXT_DATA__" type="application\/json">(.*?)<\/script>/);
if (match) {
  const data = JSON.parse(match[1]);
  const widgets = data.props.pageProps.data.widgets;
  let result = "";
  widgets.forEach(w => {
    if (w.widgetName === 'LXFaqGroup' || w.widgetName === 'LXFaqGrid' || w.widgetName === 'FaqGrid' || w.widgetName.includes('Faq')) {
      if (w.widgetValue && Array.isArray(w.widgetValue)) {
         w.widgetValue.forEach(v => {
             if (v.faqItems && Array.isArray(v.faqItems)) {
                 v.faqItems.forEach(item => {
                     result += `Q: ${item.teaserTitle}\nA: ${item.teaserText}\n\n`;
                 });
             }
         });
      }
    }
  });
  if(result) {
      fs.writeFileSync("extracted_faqs.txt", result);
  } else {
      // Just dump all string values that look like questions
      fs.writeFileSync("extracted_faqs.json", JSON.stringify(widgets, null, 2));
  }
}

File: scripts/maintenance/parse_faq.py
import re
import json

with open("/Users/ishtarpissano/.gemini/antigravity/brain/623961af-513c-49e6-b31d-d37c2d1ecdc4/.system_generated/steps/408/content.md", "r") as f:
    text = f.read()

# Find the json chunk
match = re.search(r'<script id="__NEXT_DATA__" type="application/json">([\s\S]*?)</script>', text)
if match:
    data = json.loads(match.group(1))
    
    def extract_faqs(obj):
        faqs = []
        if isinstance(obj, dict):
            # Often, essilor stores questions in teaserTitle and answers in teaserText
            if "teaserTitle" in obj and isinstance(obj["teaserTitle"], str) and "?" in obj["teaserTitle"]:
                faqs.append((obj["teaserTitle"], obj.get("teaserText", "")))
            for k, v in obj.items():
                faqs.extend(extract_faqs(v))
        elif isinstance(obj, list):
            for item in obj:
                faqs.extend(extract_faqs(item))
        return faqs

    results = extract_faqs(data)
    for q, a in results:
        print(f"Q: {q}")
        print(f"A: {a}")
        print("---")
else:
    print("No next data found")
